fix: auto generated user login 'undefined' appears on login input)[fi…

yacineahmaich opened the pull request:
fix: auto generated user login 'undefined' appears on login input)[fix: auto generated user login 'undefined' appears on login input

When We enter a lastname then we enter a firstname the generated login looks fine
Example:
    - lastname = Yacine
    - firstname = Ahmaich
    - login = yahmaich
But if we clear the firstname or the lastname we got
    - lastname = Yacine
    - firstname =  // we clear the value of firstname
    - login = yundefined

I fix the issue and also add a dot in the generated login because 'y.ahmaich' as login looks better than 'yahmaich'

yacineahmaich committed:
fix: auto generated user login 'undefined' appears on login input)[fix: auto generated user login 'undefined' appears on login input

When We enter a lastname then we enter a firstname the generated login looks fine
Example:
    - lastname = Yacine
    - firstname = Ahmaich
    - login = yahmaich
But if we clear the firstname or the lastname we got
    - lastname = Yacine
    - firstname =  // we clear the value of firstname
    - login = yundefined

I fix the issue and also add a dot in the generated login because 'y.ahmaich' as login looks better than 'yahmaich'

eldy commented on htdocs/user/card.php:
```diff
@@ -941,15 +941,19 @@
+					login = "";
+					if(lastname) {
+						if(firstname) {
+							login = firstname + ".";
```
Because the . introduce a behaviour change and may generate trouble for some users, can you replace the 
+ '.' 
with
+ "<php echo getDolGlobalString("MAIN_USER_SEPARATOR_CHAR_FOR_GENERATED_LOGIN"); ?>"
So default doe snot change and end user can decide the separator if he want by hidding the constant MAIN_USER_SEPARATOR_CHAR_FOR_GENERATED_LOGIN to the good char.

yacineahmaich committed:
feat: add a configuration constant  of separator char  for generated login.

use constant **MAIN_USER_SEPARATOR_CHAR_FOR_GENERATED_LOGIN** to specify a separator for the auto generated login
